specifications/16-strategy-variants.json

```diff
@@ -0,0 +1,118 @@
+{
+  "name": "16-strategy-variants",
+  "state": {
+    "version": 1,
+    "features": [
+      {
+        "name": "Feature.strategy.variant.with.constraint",
+        "description": "Strategy variant based on constraint",
+        "enabled": true,
+        "strategies": [
+          {
+            "name": "flexibleRollout",
+            "parameters": {
+              "rollout": "100",
+              "stickiness": "default",
+              "groupId": "a",
+              "variant": {
```

Comment:
Okay I've thought about this some more and I'm not sure this is the right place for the variant stuff (I know you're still working on this, so my thoughts might be outdated by the time you see them).

Our statically typed SDKs treat the parameters map as some kind of HashMap<String, String>, which makes this hard to ignore in the versions we have right now and causes this to trigger a parse error. If we move this property one level up to be on the strategy itself, they should all just ignore the variant property if they can't deal with it. WDYT?

Reply:
From the SDK implementation perspective and spec perspective it's no difference. The tricky part if the unleash backend that handles parameters as a blob and required no modifications. If we move variants outside of parameters then we need a new DB column + some code to route it there. So a little bit of extra work in the backend.

Reply:
Anyway I gave your suggestion a go and from the SDK standpoint it seems to be a better option